src/content/mod_configuration.cpp

```diff
@@ -237,10 +238,24 @@ void ModConfiguration::resolveDependencies()
 				mod.unsatisfied_depends.insert(optdep);
 		}
 		// if a mod has no depends it is initially satisfied
-		if (mod.unsatisfied_depends.empty())
-			satisfied.push_back(mod);
-		else
-			unsatisfied.push_back(mod);
+		if (mod.unsatisfied_depends.empty()) {
+			if (mod.name==last_mod) {
+				auto it = satisfied.begin();
+			  satisfied.insert(it, mod);
+			} else {
+				if (mod.name!=first_mod) {
+					satisfied.push_back(mod);
+				} else {
+					m_sorted_mods.push_back(mod);
+				}
+			}
+		} else {
+      if ( (mod.name!=first_mod) && (mod.name!=last_mod) ) {
+			  unsatisfied.push_back(mod);
+      } else {
+        throw ModError("Mod selected as first/last by game is not allowed to have dependencies.");
```

Comment:
This should be checked as well at a different point. If you set the first_mod mod.conf to this, it will only break if mobs is  installed and enabled:

```
name = first_mod
description = Mod which should be loaded before every other mod.
optional_depends = mobs
```

Reply:
Check moved.